test/unit/meta/src/IsArrayOrVectorTest.cpp

```diff
@@ -0,0 +1,67 @@
+/* Copyright 2021 Jiri Vyskocil
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <alpaka/meta/IsArrayOrVector.hpp>
+
+#include <catch2/catch.hpp>
+
+#include <array>
+#include <vector>
+
+
+float arrayFloat[4] = {1.0f, 2.0f, 3.0f, 4.0f};
+
+float notAnArrayFloat = 15.0f;
+float* notAnArrayFloatPointer = &notAnArrayFloat;
+std::string notAnArrayString{"alpaka"};
+
+TEST_CASE("isArrayOrVector", "[meta]")
+{
+    constexpr bool isArrayOrVectorStdArray = alpaka::meta::IsArrayOrVector<std::array<int, 10>>::value;
+    static_assert(isArrayOrVectorStdArray, "alpaka::meta::IsArrayOrVector failed!");
```

Comment:
```suggestion
    STATIC_REQUIRE(alpaka::meta::IsArrayOrVector<std::array<int, 10>>::value);
```
Catch2 can do that for you. Please apply in other places as well.

Reply:
Didn't know that. I used to work with Boost.Test so Catch2 is new to me. Thanks!